earth_enterprise/src/fusion/autoingest/StorageManager.h

```diff
@@ -164,6 +169,7 @@ template<class AssetType>
 inline void
 StorageManager<AssetType>::AddNew(const AssetKey & key, const PointerType & value) {
   std::lock_guard<std::recursive_mutex> lock(storageMutex);
+  cache.setPruneConditionVal(DetermineIfPrune());
```

Comment:
Related to my above comment, you can remove this call and just change the below line to `cache.Add(key, value, DetermineIfPrune());`